src/plugin/framegraph/frame_graph_manager.cpp

```diff
@@ -19,43 +19,86 @@
 #include "render_tasks/d3d12_rt_hybrid_task.hpp"
 #include "render_tasks/d3d12_path_tracer.hpp"
 #include "render_tasks/d3d12_accumulation.hpp"
+#include "util/log.hpp"
 
 namespace wmr
 {
 	FrameGraphManager::~FrameGraphManager()
 	{
+		LOG("Starting render task destruction.");
+
 		// Clean up the allocated frame graphs
 		for (auto& frame_graph : m_renderer_frame_graphs)
 		{
 			frame_graph->Destroy();
 			delete frame_graph;
 			frame_graph = nullptr;
 		}
+
+		LOG("Finished render task destruction.");
 	}
 
 	void FrameGraphManager::Create(wr::RenderSystem& render_system, RendererFrameGraphType initial_type, std::uint32_t initial_width, std::uint32_t initial_height) noexcept
 	{
+		LOG("Starting framegraph manager creation.");
+
 		m_width = initial_width;
 		m_height = initial_height;
 
 		m_current_rendering_pipeline_type = initial_type;
 
+		LOG("Starting to create all rendering pipelines.");
+
 		// Add required tasks to each frame graph
 		CreateDeferredPipeline();
 		CreateHybridRTPipeline();
 		CreateFullRTPipeline();
 		CreatePathTracerPipeline();
 
+		LOG("Finished creating all rendering pipelines.");
+
+		LOG("Finalizing framegraphs...");
+
 		// Set-up the rendering pipelines (frame graph configuration)
 		for (auto& frame_graph : m_renderer_frame_graphs)
 		{
 			frame_graph->Setup(render_system);
 		}
+
+		LOG("Finished finalizing framegraphs.");
+
+		LOG("Finished framegraph manager creation.");
 	}
 
 	void FrameGraphManager::SetType(RendererFrameGraphType new_renderer_frame_graph_type) noexcept
 	{
 		m_current_rendering_pipeline_type = new_renderer_frame_graph_type;
+		
+		switch (new_renderer_frame_graph_type)
+		{
+			case wmr::RendererFrameGraphType::DEFERRED:
+				LOG("Current rendering pipeline switched to deferred.");
+				break;
+			
+			case wmr::RendererFrameGraphType::HYBRID_RAY_TRACING:
+				LOG("Current rendering pipeline switched to hybrid.");
+				break;
+			
+			case wmr::RendererFrameGraphType::FULL_RAY_TRACING:
+				LOG("Current rendering pipeline switched to full ray-tracing.");
+				break;
+			
+			case wmr::RendererFrameGraphType::PATH_TRACER:
+				LOG("Current rendering pipeline switched to path tracing.");
+				break;
+			
+			case wmr::RendererFrameGraphType::RENDERING_PIPELINE_TYPE_COUNT:
+				LOG("Current rendering pipeline switched to an invalid pipeline.");
```

Comment:
A `LOGC` may be more useful here, as `m_current_rendering_pipeline_type` is already set to an invalid type.

OR set the type in this switch and just don't set it when encountering `RENDER_PIPELINE_TYPE_COUNT`

Reply:
Resolved